Mathlib/Analysis/Calculus/ImplicitContDiff.lean

```diff
@@ -0,0 +1,147 @@
+/-
+Copyright (c) 2025 Winston Yin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Winston Yin
+-/
+import Mathlib.Analysis.Calculus.Implicit
+import Mathlib.Analysis.Calculus.InverseFunctionTheorem.ContDiff
+
+/-!
+# Implicit function theorem
+
+In this file, we apply the generalised implicit function theorem to the more familiar case and show
+that the implicit function preserves the smoothness class of the implicit equation.
```

Comment:
In fact I'm very much confused by this claim that you are doing the familiar case. For me, `f (x, y) = x + y` is a typical example of the familiar case (with \partial f/\partial y = id invertible, so the solutions to f(x, y) = 0 are a graph, of the form `y = g (x)` with `g (x) = -x`). But the kernel of the derivative is not the subspace E x {0}. So I'm not convinced your formulation is the one we want to have.